### PR TITLE
 Editing datascript for Olmsted#17, Olmsted#11, scons for Olmsted#13

### PR DESCRIPTION
**Pull request by eharkins**
https://github.com/matsengrp/olmsted/issues/17

Not sure why the first two commits are supposedly different from the PR you already merged, the third commit is 0 changes, the last one is why I am making this PR

### CODE REVIEW

**Comment by eharkins** [comment]
Latest commit adds potential fix for https://github.com/matsengrp/olmsted/issues/13

File: SConstruct [hunk 1]
```diff
@@ -502,9 +502,10 @@ def add_cluster_analysis(w):
         where N is the ranking of vlads untrimmed deduplication, and M is the multiplicity of said deduplication."""
         # This option controls which sequences get joined on in the merge for the partis_seqmeta file, which has
         # orig/new names, joined on sequence from the other file
-        sources = {'--partis-seqmeta': c['partis_seqmeta'],
+	sources = {'--partis-seqmeta': c['partis_seqmeta'],
```
Comment (metasoarous) [inline SConstruct]: Are you sure this indentation is right?

File: bin/aggregate_minadcl_cluster_multiplicities.py [hunk 2]
```diff
@@ -69,11 +71,18 @@ def seqmeta_reader(filename):
                       'duplicates': d['duplicates'].split(':')})
             for d in data]
 
+def pruned_ids_reader(filename):
```
Comment (metasoarous) [inline bin/aggregate_minadcl_cluster_multiplicities.py]: @eharkins Nice set usage

File: SConstruct (hunk 1 above)
Comment (eharkins) [reply SConstruct]: Some of the indentation looks weird in git to me even though in vim it looks the same as the preceding lines.. I can try to fix

Comment (metasoarous) [reply SConstruct]: Check to see if there's a tab in there maybe?

File: bin/build_olmsted_data.py [hunk 3]
```diff
@@ -110,12 +111,21 @@ def pull_datasets(t):
...
 ]
 
-def create_node_records(tree):
+def create_node_records(tree, nt_seqs_dict, aa_seqs_dict, seqmeta_dict):
     records = []
     leaves_counter = 1
     for n in tree.traverse('postorder'):
         n.label = n.id = n.name
-        n.type = "node"
+	n.nt_seq = nt_seqs_dict[n.name]
```
Comment (metasoarous) [inline bin/build_olmsted_data.py]: @eharkins Mmm... here too. Please convert all tabs to spaces if that's what's going on.

Comment (eharkins) [reply bin/build_olmsted_data.py]: Done

File: SConstruct (hunk 1 above)
Comment (eharkins) [reply SConstruct]: Done

File: bin/build_olmsted_data.py [hunk 4]
```diff
@@ -139,22 +149,40 @@ def create_node_records(tree):
...
 
     return records
 
-def parse_tree_data(s):
+def parse_tree_data(s, nt_seqs_dict, aa_seqs_dict, seqmeta_dict):
     t = PhyloTree(s, format=1)
-    return create_node_records(t)
+    return create_node_records(t, nt_seqs_dict, aa_seqs_dict, seqmeta_dict)
+
+def parse_fasta_string(fasta_string):
```
Comment (metasoarous) [inline bin/build_olmsted_data.py]: This is cute, but should use Biopython instead.

```
from Bio import SeqIO
seqs = SeqIO.parse(filename, 'fasta')
for seqrecord in seqs:
    print "seq data", seqrecord.id, seqrecord.seq
```

Comment (eharkins) [reply bin/build_olmsted_data.py]: No longer using this parser as of latest commit

**Pull request merged**